### PR TITLE
Add Puzzletron v2 GPU quality baseline

### PR DESCRIPTION
**Pull request by j-rausch**
### What does this PR do?

Puzzletron v2 needs one small GPU baseline that exercises the same public setup and orchestration entry points as a real campaign. This PR adds that baseline and a supported non-interactive setup mode.

The GPU test creates a tiny local Qwen 3.5 model and dataset, then runs the complete lifecycle:

- setup-v2 bundle generation and compilation;
- activation and replacement scoring, MIP solving, evaluation, and AIPerf serving measurements;
- short global distillation, final evaluation and selection, and descriptor-aware reload of the selected heterogeneous checkpoint;
- report verification and a no-op resume that preserves completed artifacts.

The implementation also defines the lifecycle contracts needed by that route:

- authored configuration, scheduler attempts, and post-MIP publications have stable execution identities;
- shard workers and aggregators receive the same ordered overrides;
- stale or legacy completed work is resubmitted, while incomplete publications remain retryable;
- consolidated distillation checkpoints preserve AnyModel topology metadata, and publication failures propagate to every distributed rank;
- AIPerf preserves offline and remote-code policies by default, with strict boolean validation and an explicit compatibility opt-in.

Setup automation uses the same wizard sections, resolved configuration, rendering, and validation as interactive setup. It accepts a defaults file, campaign directory, and profile, and fails when a required value has no resolved default.

Five scoped `# nosec` exceptions document required subprocess and process-replacement boundaries. These calls use explicit argument vectors, never invoke a shell, and do not broaden the accepted command sources.

### Usage

```bash
python examples/puzzletron/puzzletron_setup_v2.py \
  --defaults /path/to/setup-v2-defaults.yaml \
  --campaign-dir /path/to/campaign \
  --profile smoke \
  --non-interactive
```

Interactive setup remains supported. Existing campaigns remain compatible when security-policy values are booleans and checkpoint overrides are lists or tuples. Security-sensitive string and numeric values are rejected instead of being interpreted by truthiness.

### Testing

The Puzzletron CPU suite covers setup, configuration resolution, orchestration identity and recovery, post-MIP publication, checkpoint metadata, security policy, and resume behavior.

The dedicated one-GPU Nox session validates the full lifecycle in the pinned CUDA 12.9 environment, including real AIPerf measurements, distillation optimizer steps, final-checkpoint reload and CUDA forward, report generation, and resume. This Nox session is a repository automation target, not a GitHub workflow. The stacked [GPU CI workflow PR](https://github.com/NVIDIA/Model-Optimizer/pull/2167) adds the repository-owned image and required-check wiring separately.

## Summary by CodeRabbit

* **New Features**
  * Added non-interactive Puzzletron setup with campaign directory, profile, and defaults support.
  * Added configurable remote-code trust and online tokenizer-resolution options.
  * Added multi-node and grouped distributed execution support.
  * Added execution identity tracking for reliable resume, reruns, and post-MIP evaluation.
  * Added Hydra-compatible configuration overrides and scientific-notation parsing.
* **Bug Fixes**
  * Improved finalization, artifact validation, failure recovery, stale-result detection, and sequence-length consistency.
* **Documentation**
  * Documented setup requirements, security defaults, and artifact-settling behavior.
* **Tests**
  * Expanded CPU, GPU, setup, orchestration, and evaluation coverage.

### CODE REVIEW

**Comment by j-rausch** [comment]
@coderabbitai review

File: tests/unit/torch/puzzletron/test_width_scenarios.py [hunk 1]
```diff
@@ -42,6 +43,131 @@
...
+    manifest = tmp_path / "manifest.json"
+    manifest.write_text(json.dumps(payload))
+    summary = tmp_path / "summary.json"
+    summary.write_text("{}\n")
+
+    assert not replacement_finalizer.finalization_marker_is_current(
+        marker,
+        manifest,
+        summary,
+    )
```
[the remark this answers is not in the document]
Comment (j-rausch) [reply tests/unit/torch/puzzletron/test_width_scenarios.py]: fixed

**Comment by j-rausch** [comment]
@coderabbitai review

**Comment by j-rausch** [comment]
@coderabbitai full review

File: tests/unit/torch/puzzletron/test_global_kd_canonical.py [hunk 2]
```diff
@@ -779,8 +777,92 @@ class Recipe(_WeightedObjectiveMixin, BaseRecipe):
...
+            18,
+            0.5,
+            {"lm_loss": 0.25},
+            best_metric_key="lm_loss",
+        )
+    assert not (tmp_path / "epoch_2_step_18" / "saving_completed").exists()
+
+
+def test_global_kd_checkpoint_publication_failure_reaches_all_ranks(tmp_path, monkeypatch):
+    import torch
```
[the remark this answers is not in the document]
Comment (j-rausch) [reply tests/unit/torch/puzzletron/test_global_kd_canonical.py]: Fixed in b49955f056

File: examples/puzzletron/README.md [hunk 3]
```diff
@@ -85,6 +85,21 @@ The defaults file is loaded only when passed explicitly and takes precedence
 over the selected profile. To expose every per-section and nested setting, use
 the advanced flow explicitly:
 
+Automation can use the same setup entry point without answering prompts. The
+defaults file must provide every required value that has no built-in default:
+
```
[the remark this answers is not in the document]
Comment (j-rausch) [reply examples/puzzletron/README.md:90]: fixed
